scripts/build/gen_handles.py

```diff
@@ -131,9 +132,14 @@ def main():
         fp.write('#include <zephyr/device.h>\n')
         fp.write('#include <zephyr/toolchain.h>\n')
         for dev in parsed_elf.devices:
+            handles = {
```

Comment:
A comment here explaining why this sorting is being done would be useful

Reply:
Done.